src/page-multi-facility/FacilityInputForm.tsx

```diff
@@ -44,14 +44,37 @@ const FacilityInputForm: React.FC = () => {
   const model = useModel();
 
   const save = () => {
+    // When saving an object to Firestore, Firestore will reject it if it is not
+    // serializable. This type enforces that what we are saving is serializable.
+    // In the future, we add new properties to the model that are not
+    // serializable, we will get a TS error below.
+    //
+    // Definition inspired by the one posted here:
+    // https://github.com/microsoft/TypeScript/issues/1897#issuecomment-338650717
+    type JsonSerializable =
+      | boolean
+      | number
+      | Date
+      | string
+      | undefined
+      | JsonSerializable[]
+      | { [key: string]: JsonSerializable };
+
+    let modelInputs: JsonSerializable = {
+      ...model[0],
+
+      // We don't need to save localeDataSource.
+      localeDataSource: undefined,
+    };
+
     saveFacility({
       id: facility?.id,
       name: facilityName || null,
       description: description || null,
       systemType: systemType || null,
-      modelInputs: JSON.parse(JSON.stringify(model))[0],
```

Comment:
@szhu - Great work on this PR.  For what it's worth, it may be possible to simplify this by avoiding the parse/stringify code and the JsonSerialization by just doing this

```javascript
modelInputs: model[0],
```

I've tested that and it works locally on my machine, but I don't have a way to test on IE 11

Reply:
TL; DR: Ignore my comment above, it actually doesn't work.  Details below if you're interested:

Actually, nevermind.  This does allow for the model to be saved, but it serializes and saves things in a way that blows up when we try to read the data back in for form population.  For example, we currently store the dates associated with planned releases as strings which turns out to be ideal because that plays nicely with our `DateInput` components.  With the suggestion above (i.e. `modelInputs: model[0],`), the planned release dates get saved as actual timestamps in Firestore which blow things up when we try to load the form on the facility information page.

Reply:
Note about the TS definition: The JsonSerializable code doesn't run at all. It's a static check guaranteeing that the variable at this point is serializable. This helps prevent this entire class of bugs from being discovered only after we try testing it!

There are some things it doesn't check for, for example whether our code deals with Dates being converted into strings, and undefined being converted into null. We can be more explicit about those by removing the Date and undefined types from the definition.

It realize that might be hard to describe the intent of that code over text -- let's do a call tomorrow and talk about it!

Reply:
@sprad and I resolved this on a call.

Here's what we did:
- Used just `model[0]`
- Placed `JSON.parse(JSON.stringify())` inside `saveFacility`, after the call to `_.pick()`

See here for further discussion: https://github.com/Recidiviz/covid19-dashboard/pull/225/files#r415377061